heuristic/delta_calculations.py

```diff
@@ -417,4 +417,127 @@ def regret_objective_function(state, employee, off_shifts, saturdays, days, L_C_
             - 10 * state.hard_vars["delta_positive_contracted_hours"].get(employee, 0)
 
             - 100 * competency_score
+    )
+
+"""
+        What is important to calculate?
+            1. deviation_from_demand vil ikke ha noe å si. Derfor unyttig å kalkulere
+           
+            8. Shift to demand burde ikke trenges ettersom vi sjekker om en ansatt jobber den dagen. 
+            9. Cover multiple demand periods er likt den over. Burde ikke trenge å sjekkes. 
+            10. More than one shift per day burde ikke trenge det heller ettersom vi sjekker et annet sted.
+            11. above/below ikke nødvendig det heller. 
+            
+            13. Dette er opp mot objektivverdi og trenger da sannsynligvis ikke å regne egen objektivverdi. Kan bare se på hver ansatt. 
+        """
+"""
+            2. weekly rest. Når vi setter inn et skift så bør denne kalkuleres. Bør derimot strengt tatt ikke sette verdien, men heller regne ut høyeste mulige verdi.
+            3. Partial weekend vil ha noe å si hvis shiftet ble satt på en lørdag eller søndag
+            4. isolated working day er vanskelig å enkelt kalkulere uten å se på dagen før og dagen etter
+            5. Samme som over. Må sjekke dagen før og dagen etter.
+            6. Consecutive days må sjekke strengt tatt L_C_D - 1 dager før og L_C_D -1 dager etter. 
+            7. Negative deviation from contracted hours. Dette går fort å regne ut. Bare ta nåværende - varighet på skift. Alt som blir minus vil da også bryte hard restriksjon.
+            12. daily_rest bør sjekkes. Bare sjekke den dagen. 
+"""
+
+def employee_shift_value(state, e, shift, saturdays, sundays, invalid_shifts, shift_combinations_violating_daily_rest, shift_sequences_violating_daily_rest, shifts_at_week, weeks, shifts_at_day, week, competency_score):
```

Comment:
Does this function calculate the cost of placing an employee on a given shift?

Reply:
I. e. does it work apriori or posteriori?

Reply:
It calculates the cost of the shift passed in as argument for the employee e. It does not set the shift.